pkg/reconciler/domainmapping/resources/certificate.go

```diff
@@ -32,5 +32,5 @@ import (
 func MakeCertificate(dm *v1alpha1.DomainMapping, certClass string) *networkingv1alpha1.Certificate {
 	certName := kmeta.ChildName(dm.GetName(), "")
 	return routeresources.MakeCertificate(
-		dm, serving.DomainMappingUIDLabelKey, dm.Name, certName, certClass)
+		dm, serving.DomainMappingUIDLabelKey, dm.Name, certName, certClass, dm.Name)
```

Comment:
Hmm -- maybe this should be empty-string for the domain-mapping use case (indicating that there isn't a base domain, and we should be using this exact name)?